### PR TITLE
PLANET-7106 Implement ESLint config in Master Theme

### PR DESCRIPTION
**Pull request by Osong-Michael**
**Description: [PLANET-7106](https://jira.greenpeace.org/browse/PLANET-7106)**

Updated ESLint config to match plugin repo

### CODE REVIEW

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [leda](https://www-dev.greenpeace.org/test-leda) | [admin](https://www-dev.greenpeace.org/test-leda/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-leda/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-leda) | [composer-local.json](https://github.com/greenpeace/planet4-test-leda/blob/main/composer-local.json)

:watch: 2023.03.29 06:34:14

File: assets/src/js/header.js [hunk 1]
```diff
@@ -89,6 +89,7 @@ const closeInactiveNavElements = event => {
 
   activeElements.forEach(button => {
     const buttonTarget = button.dataset && button.dataset.bsTarget;
+    // eslint-disable-next-line @wordpress/no-unused-vars-before-return
```
Comment (mleray) [inline assets/src/js/header.js:92]: Instead of disabling this here, we could move the `buttonTargetElement` declaration after the `return` statement

File: assets/src/js/header.js [hunk 2]
```diff
@@ -132,7 +133,10 @@ const setMobileTabsMenuScroll = () => {
   let supportsPassive = false;
   try {
     const opts = Object.defineProperty({}, 'passive', {
-      get: function() { supportsPassive = true; } // eslint-disable-line getter-return
+      get() {
```
Comment (mleray) [inline assets/src/js/header.js:136]: I think that here it should be:
```
get: () => {
   supportsPassive = true;
}
```

File: assets/src/js/header.js [hunk 3]
```diff
@@ -187,11 +191,10 @@ const setMobileTabsMenuScroll = () => {
       lastScrollDir = dir;
       lastScrollRef = lastScrollTop;
       menu.classList.remove('mobile-menu-hidden');
-      return;
```
Comment (mleray) [inline assets/src/js/header.js:190]: Why did you remove this one? There are others in the same file (lines 178 & 186) that you didn't touch

File: assets/src/js/header.js [hunk 4]
```diff
@@ -243,8 +246,10 @@ export const setupHeader = () => {
 
   let searchFocused = false;
   const searchInput = document.getElementById('search_input');
+  // eslint-disable-next-line no-unused-expressions
```
Comment (mleray) [inline assets/src/js/header.js:249]: I think that here instead of disabling ESLint we could refactor using `if` statements, such as:
```
if (searchInput) {
  ...
}
```

File: assets/src/js/menu_editor.js [hunk 5]
```diff
@@ -60,15 +62,20 @@ const menuEditorRestrictions = () => {
...
+  const disableObserver = () => {
+    observer.disconnect();
+  };
 
   /**
    * Show a notice message to inform to editors to prevent
    * using long texts to menu items
    */
   const showInfoNoticeMessage = () => {
+    // eslint-disable-next-line no-shadow
```
Comment (mleray) [inline assets/src/js/menu_editor.js:78]: Here instead of the ESLint disable we can probably just remove the line, since `menuEditorFooter` is declared earlier

File: assets/src/js/header.js (hunk 2 above)
Comment (Osong-Michael) [reply assets/src/js/header.js]: I still need to add the return statement for Sonar Cloud issues

Screenshot 2023-03-30 at 10 15 55

Comment (mleray) [reply assets/src/js/header.js]: I see that @lithrel initially added this function, I'd like his input here since I'm not sure I really understand what it does 😅

Comment (lithrel) [reply assets/src/js/header.js]: (sorry, missed the point :grin: ) So it doesn't have to return anything, it just sets the `supportsPassive` value

Comment (Osong-Michael) [reply assets/src/js/header.js]: That is fine, the return statement I added is just for tests to pass and not complain about bugs.

File: assets/src/js/header.js (hunk 3 above)
Comment (Osong-Michael) [reply assets/src/js/header.js:190]: I have removed all the others, the return statement was not necessary.

File: assets/src/js/header.js (hunk 2 above)
Comment (mleray) [reply assets/src/js/header.js]: I know, but I just wanted to make sure that it doesn't affect the function's expected behaviour?

File: assets/src/js/header.js (hunk 3 above)
Comment (mleray) [reply assets/src/js/header.js:190]: For performance it would probably better to find a way to keep doing this

File: assets/src/js/header.js (hunk 2 above)
Comment (lithrel) [reply assets/src/js/header.js]: @mleray just tested it, seems to work for me :+1:

**Review by mleray — APPROVED** [review]
Looks good to me 👍
